### DISCUSSION TITLE
Defining the motion of boundary particles

### DISCUSSION THREAD

**Discussion by Nadun19** [Xiangyu-Hu/SPHinXsys#328] [Q&A] [answered]
Hi 

I want to load in the x,y,z positions (i.e. using a csv file) of the boundary SPH particles over time. Which function should I use to accomplish this? I had a look at all the examples available and I couldn't find a suitable one. 

Thanks
Nadun

**Comment by Xiangyu-Hu** [comment]
Do you require reload the data during the simulation? If it is the case, you can write a simple dynamics in which the particle position can be updated according to read data.

**Comment by Xiangyu-Hu** [comment, marked as answer]
It seems a sub type of constraint solid dynamics.
https://github.com/Xiangyu-Hu/SPHinXsys/blob/6e9db3ab0311078972747ade4dac2c4d10d4325e/src/shared/particle_dynamics/solid_dynamics/constraint_dynamics.h#L61-L78
you can derive from this class for the proposed function.

**Reply by Nadun19** [reply]
Great thanks!
